server/leafnode_test.go

```diff
@@ -4072,3 +4099,942 @@ func TestLeafNodeInterestPropagationDaisychain(t *testing.T) {
+jetstream: { %s store_dir: %s; max_mem: 50Mb, max_file: 50Mb, extension_hint: will_extend }
+server_name: LA
+leafnodes:{
+	no_advertise: true
+    remotes:[{url:nats://a1:a1@127.0.0.1:%d, account: A},
+		     {url:nats://s1:s1@127.0.0.1:%d, account: SYS}]
+}
+# add the cluster here so we can test placement
+cluster: { name: clustL }
+`
+	for _, withDomain := range []bool{true, false} {
+		t.Run(fmt.Sprintf("with-domain:%t", withDomain), func(t *testing.T) {
+			jsDisabledDomainString := _EMPTY_
+			jsEnabledDomainString := _EMPTY_
+			if withDomain {
+				jsEnabledDomainString = `domain: "domain", `
+				jsDisabledDomainString = `domain: "domain"`
+			} else {
+				// in case no domain name is set, fall back to the extension hint.
+				// since JS is disabled, the value of this does not clash with other uses.
+				jsDisabledDomainString = "extension_hint: will_extend"
+			}
+
+			sd1 := createDir(t, JetStreamStoreDir)
+			defer os.RemoveAll(sd1)
+			confA := createConfFile(t, []byte(fmt.Sprintf(tmplA, jsEnabledDomainString, sd1)))
+			defer removeFile(t, confA)
+			sA, _ := RunServerWithConfig(confA)
+			defer sA.Shutdown()
+
+			sd2 := createDir(t, JetStreamStoreDir)
+			defer os.RemoveAll(sd2)
+			confB := createConfFile(t, []byte(fmt.Sprintf(tmplB, jsEnabledDomainString, sd2)))
+			defer removeFile(t, confB)
+			sB, _ := RunServerWithConfig(confB)
+			defer sB.Shutdown()
+
+			confC := createConfFile(t, []byte(fmt.Sprintf(tmplC, jsDisabledDomainString)))
+			defer removeFile(t, confC)
+			sC, _ := RunServerWithConfig(confC)
+			defer sC.Shutdown()
+
+			checkClusterFormed(t, sA, sB, sC)
+			c := cluster{t: t, servers: []*Server{sA, sB, sC}}
+			c.waitOnPeerCount(2)
+
+			sd3 := createDir(t, JetStreamStoreDir)
+			defer os.RemoveAll(sd3)
+			// deliberately pick server sC (no JS) to connect to
+			confLA := createConfFile(t, []byte(fmt.Sprintf(tmplLA, jsEnabledDomainString, sd3, sC.opts.LeafNode.Port, sC.opts.LeafNode.Port)))
+			defer removeFile(t, confLA)
+			sLA, _ := RunServerWithConfig(confLA)
+			defer sLA.Shutdown()
+
+			checkLeafNodeConnectedCount(t, sC, 2)
+			checkLeafNodeConnectedCount(t, sLA, 2)
+			c.waitOnPeerCount(3)
+			peers := c.leader().JetStreamClusterPeers()
+			for _, peer := range peers {
+				if _, ok := expectedJetStreamPeers[peer]; !ok {
+					t.Fatalf("Found unexpected peer %q", peer)
+				}
+			}
+
+			// helper to create stream config with uniqe name and subject
+			cnt := 0
+			strmCfg := func(placementCluster string) *nats.StreamConfig {
+				name := fmt.Sprintf("s-%d", cnt)
+				cnt++
+				if placementCluster == "" {
+					return &nats.StreamConfig{Name: name, Replicas: 1, Subjects: []string{name}}
+				}
+				return &nats.StreamConfig{Name: name, Replicas: 1, Subjects: []string{name},
+					Placement: &nats.Placement{Cluster: placementCluster}}
+			}
+
+			test := func(port int, expectedDefPlacement string) {
+				ncA := natsConnect(t, fmt.Sprintf("nats://a1:a1@127.0.0.1:%d", port))
```

Comment:
I think `defer ncA.Close()` is missing here.

Reply:
fixed